sklearn/tree/_splitter.pyx

```diff
@@ -539,6 +539,10 @@ cdef inline int node_split_best(
             best_split.impurity_right
         )
 
+        # Note: this should always be called at the very end because it will
+        # move samples around, thereby affecting the criterion.
+        # This affects the computation of the children impurity, which affects
+        # the computation of the next node.
```

Comment:
Not sure that comment is useful because the name of the function is quite explicite. One possibilty is to add this comment in the docstring of the function if there not enough details there.

Reply:
Moved into the docstring of the function